packages/twitch-chat-client/src/ChatClient.ts

```diff
@@ -1372,10 +1372,11 @@ export class ChatClient extends IrcClient {
 						message === 'Improperly formatted AUTH' ||
 						message === 'Invalid NICK'
 					) {
-						this._authVerified = false;
 						this._authFailureMessage = message;
 						this.emit(this.onAuthenticationFailure, message);
-						this._connection!.disconnect();
+						// Attempt to reconnect right away if auth was previously valid, else wait 5 seconds to avoid spamming
+						setTimeout(async () => this.reconnect(), this._authVerified ? 0 : 5000);
```

Comment:
I would prefer that this uses the reconnect capability of the lower level connection, if possible, in order to make use of its gradual increase in retry delay.

Reply:
That's what I was trying to do at first, but the connection interface does not have reconnect: https://github.com/d-fischer/connection/blob/master/src/Connection.ts

That being said, the higher level reconnect does end up calling `disconnect`, then `connect` on the underlying connection, so I don't really see a tangible difference here.

High level reconnect we are using:
https://github.com/d-fischer/ircv3/blob/afed4389b49d973ccaced56b7935a2d3f591fc65/src/IrcClient.ts#L410-L413

Inside that quit command:
https://github.com/d-fischer/ircv3/blob/afed4389b49d973ccaced56b7935a2d3f591fc65/src/IrcClient.ts#L610

Inside that connect command:
https://github.com/d-fischer/ircv3/blob/afed4389b49d973ccaced56b7935a2d3f591fc65/src/IrcClient.ts#L436

What the `PersistentConnection` reconnect does:
https://github.com/d-fischer/connection/blob/c1db58af8633e52cb264e2b91b17320457da9d56/src/PersistentConnection.ts#L120-L123

Reply:
I just checked, and at the point where you submit your password, the connection is considered "connected" already and can't profit from the gradual backoff anyway.

In any case, there *should* be gradual backoff for password authentication as well, to guarantee a fast reconnect if the refresh works and still prevent spamming if it does not, so it might make sense to use a similar technique here.